### ISSUE TITLE
Stop the Moving Window

### ISSUE THREAD

**Issue by ax3l** [ECP-WarpX/WarpX#1441]
@hightower8083 asked on Gitter about a functionality to stop the moving window after a certain time step.

He currently tries to achieve that with the `Physics_applications/plasma_acceleration/inputs_2d` example, restarting at a certain time step, and then re-starting with a changed input file with `warpx.do_moving_window` changed to `0`, which is not successful (fields are off).

Looks similar to me like:
- https://github.com/ComputationalRadiationPhysics/picongpu/pull/1668
- https://github.com/ComputationalRadiationPhysics/picongpu/pull/2792

@hightower8083 please feel free to add more details and/or the exact modified input scripts you executed, please.

**Comment by hightower8083** [comment]
yes the idea is to deactivate the `moving_window` via sectioning the run with checkpoint/restart and use a modified input script for the restart. 
For a  minimal example one take `plasma_acceleration/inputs_2d` and modify the corresponding lines:
```python
plasma_e.zmax = 1.e-2       # for decorative purposes

diagnostics.diags_names = diag1 diag_rstrt
diag1.period = 100
diag1.diag_type = Full
diag1.format = openpmd     # format choice is also optional
diag1.openpmd_backend = h5

diag_rstrt.period = 1000
diag_rstrt.format = checkpoint
diag_rstrt.diag_type = Full
```

For restart, one can use same script and modify it as following:
```python
stop_time = 4e-12
amr.restart = diags/diag_rstrt00957
warpx.do_moving_window = 0

# Diagnostics
diagnostics.diags_names = diag1
diag1.period = 10    # for decorative purposes
diag1.diag_type = Full
diag1.format = openpmd
diag1.openpmd_backend = h5
```

Considering the `particles` diagnostics one can see that windows does get stopped and simulation proceeds as may be expected
![anim1](https://user-images.githubusercontent.com/22051570/96022806-b95dbc80-0e51-11eb-9378-e29f416381a5.gif)

however the field output is wrong -- here is the fields before and right after the restart:

![](https://files.gitter.im/5d795645d73408ce4fcaa804/QeCt/image.png)

the field array shape (grid) before restart is (128, 64) becomes (2, 64)

**Comment by ax3l** [comment]
As a quick-hack, try to execute this block unconditionally (`if(true)`) for the executable used during restart:
https://github.com/ECP-WarpX/WarpX/blob/c0ee57664cedc31a5c16c809482e3068828a7755/Source/Diagnostics/Diagnostics.cpp#L222-L229

**Comment by hightower8083** [comment]
I think for a moment i'll better focus on the #1442 rather than on the hacking the restart, although IMHO having the window modifiable at restart could be a great tool allowing more flexibility for the  composite cases

**Comment by ax3l** [comment]
Implemented in #2027
